src/client/dfs/obj.c

```diff
@@ -852,8 +852,10 @@ ostatx_cb(tse_task_t *task, void *data)
 		D_GOTO(out, rc = daos_errno2der(rc));
 
 	if (S_ISREG(args->obj->mode)) {
-		args->stbuf->st_size   = op_args->array_stbuf.st_size;
-		args->stbuf->st_blocks = (args->stbuf->st_size + (1 << 9) - 1) >> 9;
+		args->stbuf->st_size    = op_args->array_stbuf.st_size;
+		args->stbuf->st_blocks  = (args->stbuf->st_size + (1 << 9) - 1) >> 9;
+		args->stbuf->st_blksize = op_args->entry.chunk_size ? op_args->entry.chunk_size :
```

Comment:
we do say that we return this as 0 in the daos_fs.h header, so you will need to update that there.
also if you are setting this in ostatx, it does not make sense we do not set it elsewhere, so we need to change entry_stat() also to set this for other stat calls, otherwise we introduce inconsistency in what API is being used.

Reply:
st_blksize is set currently inside st_blksize().
https://github.com/daos-stack/daos/blob/master/src/client/dfs/common.c#L553
@mchaarawi Could you please let me know the place in daos_fs.h needing changes? Thank you!

Reply:
ah indeed i missed that